x-pack/filebeat/input/awss3/state.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package awss3
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// State is used to communicate the reading state of a file
+type State struct {
+	Id           string    `json:"id" struct:"id"`
+	Bucket       string    `json:"bucket" struct:"bucket"`
+	Key          string    `json:"key" struct:"key"`
+	LastModified time.Time `json:"last_modified" struct:"last_modifed"`
+	Size         int64     `json:"size" struct:"size"`
+	Stored       bool      `json:"stored" struct:"stored"`
+}
+
+// NewState creates a new s3 object state
+func NewState(bucket, key string, size int64, lastModified time.Time) State {
+	s := State{
+		Bucket:       bucket,
+		Key:          key,
+		LastModified: lastModified,
+		Size:         size,
+		Stored:       false,
+	}
+
+	s.Id = s.Bucket + s.Key + strconv.FormatInt(s.Size, 10) + s.LastModified.String()
```

Comment:
Have you considered using the ETag rather than size and last modified? The ETag should only change when the content changes which could make it useful as part of a unique ID.

Reply:
much better, yes, I will change to `Etag` for equalness, `LastMofidied` we still need for states tracking